filebeat/crawler/prospector_stdin.go

```diff
@@ -15,18 +13,15 @@ type ProspectorStdin struct {
 	started    bool
 }
 
-func NewProspectorStdin(config cfg.ProspectorConfig, channel chan *input.FileEvent) (*ProspectorStdin, error) {
+func NewProspectorStdin(p *Prospector) (*ProspectorStdin, error) {
 
-	prospectorer := &ProspectorStdin{}
+	prospectorer := &ProspectorStdin{
+		Prospector: p,
```

Comment:
Perhaps :-) Currently it simplifies things a lot. But I could see that this dependency eliminates itself with further refactoring.